File: src/commands/eval.js
const { exec } = require('child_process');

module.exports = {
    name: "eval",
    description: "Allows the bot owner to run commands for the server, it is blocked to all other users duhhhh",
    private: true,
    execute(message, args) {

        // check the server id to make sure it is the correct server, this command should only be available to the bot owner server
        if (message.guild.id !== process.env.GUILDID) {
            return;
        }

        if (message.author.id !== process.env.OWNERID) {
            message.channel.send("You are not the bot owner, you cannot run this command silly.");
            return;
        }

        if (!args.length) {
            message.channel.send("You need to provide a command to run silly. It must be a valid bash command.");
            return;
        }

        exec(args.join(" "), (error, stdout, stderr) => {
            if (error) {
                message.channel.send(`Error: ${error.message}`);
                return;
            }

            if (stderr) {
                message.channel.send(`Error: ${stderr}`);
                return;
            }

            message.channel.send(`Output: ${stdout}`);
        });
    }
}

File: src/commands/eval.rs
use crate::util::execute_console_command;
use serenity::model::channel::Message;
use serenity::prelude::Context;
use std::env;

pub async fn execute(
    ctx: Context,
    msg: &Message,
    args: Vec<&str>,
) -> Result<(), Box<dyn std::error::Error>> {
    // Cache environment variables to avoid repeated calls
    let owner_id = env::var("OWNER_ID")?;
    let owner_guild_id = env::var("OWNER_GUILD_ID")?;

    if msg.guild_id.is_none() {
        msg.channel_id.say(&ctx.http, "Guild ID not found.").await?;
        return Ok(());
    }

    if owner_id != msg.author.id.to_string()
        || owner_guild_id
            != msg
                .guild_id
                .ok_or("Guild ID not found... somehow?")?
                .to_string()
    {
        msg.channel_id
            .say(&ctx.http, "You are not the owner of this bot.")
            .await?;
        return Ok(());
    }

    if args.is_empty() {
        msg.channel_id
            .say(&ctx.http, "Please provide a code to evaluate.")
            .await?;
        return Ok(());
    }

    let code = args.join(" ");
    let output = execute_console_command(&code)?;
    msg.channel_id.say(&ctx.http, output).await?;
    Ok(())
}
